registry/api/errcode/errors.go

```diff
@@ -69,20 +69,28 @@ func (ec *ErrorCode) UnmarshalText(text []byte) error {
 // WithDetail creates a new Error struct based on the passed-in info and
 // set the Detail property appropriately
 func (ec ErrorCode) WithDetail(detail interface{}) Error {
```

Comment:
@endophage do you mean that you would expect the input type to match the return type?
If so, in order to do that then I think we'd have to replace the ErrorCode type of just Error.
Or are you suggesting that we simply rename "WithXXX" to something else so it doesn't confuse people who are used to Go's WithXXX methods?